### PR TITLE
daml-on-x-server: document all participant state interfaces

### PR DESCRIPTION
**Pull request by meiersi-da**
This PR solves the first half of #138 as explained in its description.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/ReadService.scala [hunk 1]
```diff
@@ -6,33 +6,155 @@ package com.daml.ledger.participant.state.v1
...
+    * Last but not least, there is an expectation about the relation between streams visible
+    * on *separate* participant state implementations connected to the same ledger.
+    * The expectation is that two parties hosted on separate participant nodes are in sync
+    * on transaction nodes and contracts that they can both see. The more formal definition
+    * is based on the notion of projections of transactions
+    * (see https://docs.daml.com/concepts/ledger-model/ledger-privacy.html), as follows.
+    *
+    * Assume that there is
+    * - a party `A` hosted at participant `p1`,
+    * - a party * `B` hosted at participant `p2`, and
```
Comment (gleber-da) [inline ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/ReadService.scala]: Typo `*`
```suggestion
    * - a party `B` hosted at participant `p2`, and
```

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala [hunk 2]
```diff
@@ -3,7 +3,93 @@
...
+  *
+  * The methods in this interface are all methods that are supported
+  * *uniformly* across all ledger participant implementations. Methods for
+  * uploading packages, on-boarding parties, and changing ledger-wide
+  * configuration are specific to a ledger and therefore to a participant
+  * implementation. Moreover, these methods usually require admin-level
+  * privileges, whose granting is also specific to a ledger.
+  *
+  * If a ledger is run for testing only, there is the option for quite freely
+  * allowing the on-boarding
```
Comment (gleber-da) [inline ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala]: Please reflow text here.

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala [hunk 3]
```diff
@@ -3,7 +3,93 @@
...
+  *
+  */
 trait WriteService {
+
+  /** Submit a transaction for acceptance to the ledger.
+    *
+    * This method must be thread-safe, not throw, and not block on IO. It is
+    * though allowed to perform significant computation. The expectation is
+    * that callers call this method on a thread dedicated to getting this transaction
+    * ready for acceptance to the ledger,which typically requires some
```
Comment (gleber-da) [inline ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala]:
```suggestion
    * ready for acceptance to the ledger, which typically requires some
```

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/ReadService.scala (hunk 1 above)
Comment (meiersi-da) [reply ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/ReadService.scala]: fixed by @dajmaki

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala (hunk 2 above)
Comment (meiersi-da) [reply ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala]: done

File: ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala (hunk 3 above)
Comment (meiersi-da) [reply ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala]: done

**Pull request merged**